### PR TITLE
Parking size tests

### PR DESCRIPTION
**Pull request by rominavaliunas**

### CODE REVIEW

File: app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java [hunk 1]
```diff
@@ -1,18 +1,62 @@
...
+    @Test
+    public void onParkingSizeCreationButtonPressed_invalidSize() {
+        //Given
+        when(view.getSizeSubmitted()).thenReturn("999999999999999999");
+        doThrow(new NumberFormatException()).when(model).setParkingSize("999999999999999999");
+        //When
+        presenter.onParkingSizeCreationButtonPressed();
+        //Then
+        verify(view).showInvalidSizeError();
+    }
```
Comment (imachadoglobant) [inline app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java:56]: Please also validate size is greater than 0 and add a test for it.

**Review by imachadoglobant — CHANGES_REQUESTED** [review]

File: app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        when(model.getParkingSize()).thenReturn(15);
+        //When
+        presenter.onParkingSizeCreationButtonPressed();
+        //Then
+        verify(model).setParkingSize("15");
+        verify(view).showParkingSize(15);
+    }
+
+    @Test
+    public void onParkingSizeCreationButtonPressed_invalidSize() {
```
Comment (carlos-navas-globant) [inline app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java]: _methodName_condition_result_

I think you are missing the result in the name

File: app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @Before
+    public void setup() {
+        model = mock(ParkingModel.class);
+        view = mock(ParkingView.class);
+        presenter = new ParkingPresenter(model, view);
+        logMockedStatic = mockStatic(Log.class);
+    }
+
+    @Test
+    public void onParkingSizeCreationButtonPressed_validSize() {
```
Comment (carlos-navas-globant) [inline app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java]: methodName_condition_result

I think you are missing the result in the name

**Review by carlos-navas-globant — CHANGES_REQUESTED** [review]

File: app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java (hunk 2 above)
Comment (rominavaliunas) [reply app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java]: names updated!

File: app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java (hunk 3 above)
Comment (rominavaliunas) [reply app/src/test/java/com/example/parkingsystem/ParkingSizeTest.java]: names updated!

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java [hunk 4]
```diff
@@ -17,8 +17,12 @@ public ParkingPresenter(ParkingModel newParkingModel, ParkingView newParkingView
 
     public void onParkingSizeCreationButtonPressed() {
         try {
-            parkingModel.setParkingSize(parkingView.getSizeSubmitted());
-            parkingView.showParkingSize(parkingModel.getParkingSize());
+            if ((Integer.parseInt(parkingView.getSizeSubmitted())) == 0) {
```
Comment (imachadoglobant) [inline app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Here you are parsing parking size twice, once in ParkingModel.setParkingSize and another time here. You should better move this check to ParkingModel and throw an IllegalArgumentException when size is equal or less than 0. Then you can catch here, just as NumberFormatException.

**Review by imachadoglobant — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java (hunk 4 above)
Comment (rominavaliunas) [reply app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java]: Added IllegalArgumentException

**Review by imachadoglobant — APPROVED** [review]

**Pull request merged**